cinema/views.py

```diff
@@ -101,8 +105,24 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminOrIfAuthenticatedReadOnly],
+        url_path="upload-image",
```

Comment:
Maybe the IsAdminUser permission class fits better here?
Authenticated users cant use the POST method anyway